Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.py

```diff
@@ -0,0 +1,760 @@
+        self.refresh_token = refresh_token
+        self.enc_key = enc_key
+        self.use_ssl = use_ssl
+        self._get_access_token()
+
+    def _get_access_token(self) -> str:
+        """
+        Checks if access token exists in the integration context and return it if it exists, if not, a new token
+        is generated and saved in the integration context along with the query api_url and the instance_id
+        Returns:
+            The access token from the integration context or from the request.
+        """
+        integration_context = demisto.getIntegrationContext()
+        access_token = integration_context.get(ACCESS_TOKEN_CONST)
+        valid_until = integration_context.get(EXPIRES_IN)
+        if access_token and valid_until:
+            if int(time.time()) < valid_until:
+                return access_token
+        access_token, api_url, instance_id, refresh_token, expires_in = self._oproxy_authorize()
+        if not access_token or not instance_id:
+            raise DemistoException('Could not get access token and instance_id')
+        updated_integration_context = {
+            ACCESS_TOKEN_CONST: access_token,
+            EXPIRES_IN: int(time.time()) + expires_in - 30,
+            API_URL_CONST: api_url,
+            INSTANCE_ID_CONST: instance_id
+        }
+        if refresh_token:
+            updated_integration_context.update({REFRESH_TOKEN_CONST: refresh_token})
+        demisto.setIntegrationContext(updated_integration_context)
+        return access_token
+
+    def _oproxy_authorize(self):
+        oproxy_response = self._http_request('POST',
+                                             '/cdl-token',
+                                             json_data={'token': get_encrypted(self.refresh_token, self.enc_key)})
+        access_token = oproxy_response.get(ACCESS_TOKEN_CONST)
+        api_url = oproxy_response.get('url')
+        refresh_token = oproxy_response.get(REFRESH_TOKEN_CONST)
+        instance_id = oproxy_response.get(INSTANCE_ID_CONST)
+        expires_in = int(oproxy_response.get(EXPIRES_IN, 60 * 60))
+        return access_token, api_url, instance_id, refresh_token, expires_in
+
+    def query_loggings(self, query: str) -> List[Dict[str, Any]]:
+        """
+        This function handles all the querying of Cortex Logging service
+
+        Args:
+            query: The sql string query.
+
+        Returns:
+            A list of records according to the query
+        """
+        query_data = {'query': self.add_instance_id_to_query(query),
+                      'language': 'csql'}
+        query_service = self.initial_query_service()
+        response = query_service.create_query(query_params=query_data, enforce_json=True)
+        query_result = response.json()
+
+        if not response.ok:
+            status_code = response.status_code
+            message = query_result[-1].get('context')
```

Comment:
check its a list of length at least 1 and it contains dict

Reply:
Handled differently